iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/io/CompactionTsFileOutput.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.storageengine.dataregion.compaction.io;
+
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.storageengine.dataregion.compaction.schedule.CompactionTaskManager;
+import org.apache.iotdb.tsfile.write.writer.TsFileOutput;
+
+import com.google.common.util.concurrent.RateLimiter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public class CompactionTsFileOutput extends OutputStream implements TsFileOutput {
+
+  private TsFileOutput output;
+  private RateLimiter rateLimiter = CompactionTaskManager.getInstance().getMergeWriteRateLimiter();
+  private final long maxSizePerWrite;
+
+  public CompactionTsFileOutput(TsFileOutput output) {
+    this.output = output;
+    maxSizePerWrite =
+        (int)
+            (rateLimiter.getRate()
+                / IoTDBDescriptor.getInstance().getConfig().getCompactionThreadCount());
```

Comment:
And what is the point of the parameter?
If the rate is 40MB/s, the thread count is 10, and the write size is 8MB, it is totally OK.
The write will just take 2 seconds to wait for permission.

Reply:
The write will obtain write size twice, but it won't wait because the rate is 40MB/s. This parameter intends to avoid large write blocks small write: If a thread acquired large write size in one time, other threads may be blocked for a while. However, it may split a write to many part and i'm not sure what the impact is. 
Another implementation I thought is to remove `/ IoTDBDescriptor.getInstance().getConfig().getCompactionThreadCount()`.

Reply:
I do not see the point. What is so different between waiting for 1 second twice and 2 seconds once?
The total waiting time is not affected by the block size.
Blocking other threads is also quite normal, because you are doing RATE LIMITING and blocking is how exactly things are done.